### PR TITLE
osd: remove redudant call of heartbeat_check

### PR DESCRIPTION
**Pull request by liupan1111**
…dy called in tick_without_osd_lock.

Signed-off-by: Pan Liu <pan.liu@istuary.com>

### CODE REVIEW

**Comment by liupan1111** [comment]
@liewegas , please help take a look. thanks.

**Comment by yuyuyu101** [comment]
@liupan1111 plz remove extra rocksdb submodule updaate

**Comment by liupan1111** [comment]
@yuyuyu101, thanks a lot for your remind. I just revert the changes for rocksdb. Please take a look.

**Comment by yuyuyu101** [comment]
@liewegas @tchaikov I give a look at another heartbeat_check, it's called under is_active and is_waiting_for_healthy. I'm concerning IS_BOOTING state also needs heartbeat_check... Although I don't know whether it's problem...

**Comment by liupan1111** [comment]
@tchaikov, any comment about this pr?

**Comment by tchaikov** [comment]
> I give a look at another heartbeat_check, it's called under is_active and is_waiting_for_healthy. I'm concerning IS_BOOTING state also needs heartbeat_check... Although I don't know whether it's problem...

this `heartbeat_check()` call is scheduled in the `tick_without_osd_lock()` with the interval of `osd_heartbeat_interval`, with its own thread.

while the `heartbeat_check()` call to be removed lives in `osd->heartbeat_entry()`, which is in turn called in `heartbeat_thread` at the interval of `0.5 + random(0.. osd_heartbeat_interval)`, with its own thread also.

both threads are started right after the monclient starts, and before the osdmap is read from disk. so removing the `heartbeat_check()` call in `heartbeat()` won't interfere with `IS_BOOTING` or the boot process. however, my only concern is, prior to this change, we check the heartbeat with a random interval between 0.5 and `osd_heartbeat_interval`. and after this change, we check the heartbeat with a fixed interval of `osd_heartbeat_interval`.

but since the default `osd_heartbeat_grace` (20) is far larger than the default `osd_heartbeat_interval` (6), i think this change does not impact the behavior a lot. moreover, we have `osd_fast_fail_on_connection_refused`. so i think it's a safe change.

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -4296,9 +4296,6 @@ void OSD::heartbeat()
 						     now));
   }
 
-  dout(30) << "heartbeat check" << dendl;
-  heartbeat_check();
```
Comment (tchaikov) [inline src/osd/OSD.cc]: just out of curiosity, why not removing the call in `tick_without_osd_lock()`?

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2016-11-28_20:50:22-rados-wip-yuri-testing_2016_11_28-distro-basic-smithi/#

**Pull request merged**